emcc.py

```diff
@@ -934,8 +939,16 @@ def detect_fixed_language_mode(args):
                 break
             if found: break
           if found: break
-        if not found and lib not in ['GL', 'GLU', 'glut', 'm', 'c', 'SDL', 'stdc++', 'pthread']: # whitelist our default libraries
-          logging.warning('emcc: cannot find library "%s"', lib)
+        if not found:
+          system_js_libraries += shared.Building.path_to_system_js_libraries(lib)
+
+      # Certain linker flags imply some link libraries to be pulled in by default.
```

Comment:
i still feel these lines are nicer in something like `shared.Building.add_default_system_libraries_from_settings_changes` (or a better name ;) ), but i don't feel strongly about it.